etcdserver/storage.go

```diff
@@ -109,14 +108,6 @@ func readWAL(waldir string, snap walpb.Snapshot) (w *wal.WAL, id, cid types.ID,
 // It must ensure that, after upgrading, the most recent version is present.
 func upgradeDataDir(baseDataDir string, name string, ver version.DataDirVersion) error {
 	switch ver {
-	case version.DataDir0_4:
-		plog.Infof("converting v0.4 log to v2.0")
-		err := migrate.Migrate4To2(baseDataDir, name)
-		if err != nil {
-			plog.Fatalf("failed to migrate data-dir (%v)", err)
-			return err
-		}
-		fallthrough
```

Comment:
could we keep this case and fatal here? So user who run 2.2 using 0.4 data dir will not start with empty data dir and be confused about data loss.